src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -75,7 +75,6 @@ public AuthorizeFilter(AuthorizationPolicy policy)
 
             // Note: Default Anonymous User is new ClaimsPrincipal(new ClaimsIdentity())
             if (httpContext.User == null ||
```

Comment:
I'm pretty sure @Tratcher told me once that this property couldn't be `null`, but maybe things have changed.